### DISCUSSION TITLE
feat(medusa): Custom seed.json file for custom models

### DISCUSSION THREAD

**Discussion by herawais** [medusajs/medusa#4659] [Feature Requests]
Hi 
There should be a feature for seeding our custom database models, instead of the default ones,

**Current Status:**
Currently only [these](https://github.com/medusajs/medusa/blob/develop/packages/medusa/src/commands/seed.ts) models are supported , 

**Required Feature:**
but there should be a way to define custom models based on the keys in [seed.json](https://github.com/medusajs/medusa-starter-default/blob/master/data/seed.json) file

**Comment by Difenter** [comment]
@herawais Hello. Did you find the answer to your question? I'm also looking for an answer to this question(

**Reply by herawais** [reply]
unfortunately not,

**Reply by Difenter** [reply]
@herawais  thank you very much for your response. The only option I can imagine is when we want to make our demo data or just data at the beginning of the application. The answer is to use data insertion migrations.

**Reply by imash96** [reply]
Use medusa v2 and write or modify the seed.ts as per your need.

In medusa v2 you can find the seed script file under /src/scripts/seed.ts

**Reply by Difenter** [reply]
Mr. @imash96 , I have the seed file. For example, I created several new models (custom models). Can I actually write or create a new source file and add my own model?
  "size-measurement": [
    {
    "category": "top",
    "size": "XS",
    "chest": "83–86",
    "waist": "71–74",
    "hip": "82–85"
    }
  ],
  
  When I add my size dimension object to the default source file, it doesn't happen and my instance is not created.
   In the bottom, i write my package config file.
   {
  "name": "dozavtra-server",
  "version": "0.1.0",
  "description": "eCommerce server for Dozavtra shop",
  "license": "UNLICENSED",
  "keywords": [
    "sqlite",
    "postgres",
    "typescript",
    "ecommerce",
    "headless",
    "medusa"
  ],
  "scripts": {
    "clean": "cross-env ./node_modules/.bin/rimraf dist",
    "build": "cross-env npm run clean && npm run build:server && npm run build:admin",
    "build:server": "cross-env npm run clean && tsc -p tsconfig.server.json",
    "build:admin": "cross-env medusa-admin build",
    "watch": "cross-env tsc --watch",
    "test": "cross-env jest",
    "seed": "cross-env medusa seed -f ./data/seed.json",
    "start": "cross-env npm run build && medusa start",
    "start:staging": "cross-env NODE_ENV=staging npm run build && cross-env NODE_ENV=staging medusa start",
    "start:custom": "cross-env npm run build && node --preserve-symlinks --trace-warnings index.js",
    "dev": "cross-env npm run build:server && medusa develop",
    "build:image": "docker build -t dozavtra-server:$(node -p -e \"require('./package.json').version\") ."
  },
  "dependencies": {
    "@babel/cli": "^7.14.3",
    "@babel/core": "^7.14.3",
    "@medusajs/admin": "7.1.11",
    "@medusajs/cache-inmemory": "^1.8.9",
    "@medusajs/cache-redis": "^1.8.9",
    "@medusajs/event-bus-local": "^1.9.8",
    "@medusajs/event-bus-redis": "^1.8.11",
    "@medusajs/file-local": "^1.0.3",
    "@medusajs/medusa": "1.20.2",
    "@tanstack/react-query": "4.22.0",
    "@types/cors": "^2.8.17",
    "babel-preset-medusa-package": "^1.1.19",
    "body-parser": "^1.19.0",
    "cors": "^2.8.5",
    "dotenv": "16.3.1",
    "express": "^4.17.2",
    "livr": "^2.7.1",
    "medusa-fulfillment-manual": "^1.1.39",
    "medusa-interfaces": "^1.3.8",
    "medusa-payment-manual": "^1.0.24",
    "medusa-payment-stripe": "^6.0.7",
    "prism-react-renderer": "^2.0.4",
    "typeorm": "^0.3.16"
  },
  "devDependencies": {
    "@babel/preset-typescript": "^7.21.4",
    "@medusajs/medusa-cli": "^1.3.21",
    "@types/express": "^4.17.13",
    "@types/jest": "^27.4.0",
    "@types/node": "^17.0.8",
    "cross-env": "^7.0.3",
    "eslint": "^6.8.0",
    "jest": "^27.3.1",
    "rimraf": "^3.0.2",
    "ts-jest": "^27.0.7",
    "ts-loader": "^9.2.6",
    "typescript": "^4.5.2"
  },
  "jest": {
    "globals": {
      "ts-jest": {
        "tsconfig": "tsconfig.spec.json"
      }
    },
    "moduleFileExtensions": [
      "js",
      "json",
      "ts"
    ],
    "testPathIgnorePatterns": [
      "/node_modules/",
      "<rootDir>/node_modules/"
    ],
    "rootDir": "src",
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(ts|js)$",
    "transform": {
      ".ts": "ts-jest"
    },
    "collectCoverageFrom": [
      "**/*.(t|j)s"
    ],
    "coverageDirectory": "./coverage",
    "testEnvironment": "node"
  },
  "overrides": {
    "@medusajs/admin-ui": {
      "@medusajs/ui-preset": "^1.1.2"
    }
  },
  "resolutions": {
    "@medusajs/ui-preset": "^1.1.2"
  }
}
